packages/core/src/tracing/instrumentation/cloud/aws-sdk/v3/index.js

```diff
@@ -81,14 +82,23 @@ function shimSmithySend(originalSend) {
     const command = smithySendArgs[0];
 
     if (isActive) {
-      const awsProduct = operationMap[command.constructor.name];
+      const serviceId = self.config && self.config.serviceId;
+      const awsProduct = serviceId && awsProducts.find(aws => aws.spanName === serviceId.toLowerCase());
 
-      if (awsProduct) {
+      if (awsProduct && awsProduct.supportsOperation(command.constructor.name)) {
         return awsProduct.instrumentedSmithySend(self, originalSend, smithySendArgs);
+      } else {
+        // This code will be removed when we release support for version 3x.
+        // We are keeping it here for now to avoid any unknown problems.
+        handleLegacy(originalSend, smithySendArgs);
       }
-
-      return originalSend.apply(self, smithySendArgs);
     }
     return originalSend.apply(self, smithySendArgs);
```

Comment:
Great! I'll retain the existing code as-is in the 'else' section to prevent potential future bugs. Thank you all for bringing this issue to my attention 💯

Reply:
Good call. I think that is easier to read. 👍